langchain/document_loaders/blob_loaders/schema.py

```diff
@@ -0,0 +1,129 @@
+import contextlib
+import mimetypes
+from abc import ABC, abstractmethod
+from io import BytesIO
+from pathlib import PurePath
+from pydantic import BaseModel
+from typing import Union, Optional, Generator
+
+PathLike = Union[str, PurePath]
+
+
+class Blob(BaseModel):
+    """A blob is used to represent raw data by either reference or value.
+
+    Provides an interface to materialize the blob in different representations.
+
+    This is inspired by: https://developer.mozilla.org/en-US/docs/Web/API/Blob
+
+    """
+
+    data: Union[bytes, str, None]
+    mimetype: Optional[str] = None
+    encoding: str = "utf-8"  # Use utf-8 as default encoding, if decoding to string
+    # Location where the original content was found
+    # Represent location on the local file system
+    # Useful for situations where downstream code assumes it must work with file paths
+    # rather than in-memory content.
+    path: Optional[PathLike] = None
```

Comment:
So if this was pulled from e.g., a URL this would be `None`? A temp dir?

Reply:
Undecided -- we can set to None or set to the source URL if known or set to a temp location on disk if it was downloaded to the file system and not stored in memory.

Any opinions?

We can extend it to support URLs and support a driver for loading content feels a bit complex..

e.g., there's more than one way to fetch an HTML file (one with requests and another with something like playwright to execute the js)

Reply:
I don't have opinions here. You're right, that sounds complex